interchain-token-factory/src/proxy.rs

```diff
@@ -0,0 +1,236 @@
+    ) {
+        self.token_manager_proxy(sc_address)
+            .transfer_mintership(minter)
+            .execute_on_dest_context::<()>();
+    }
+
+    fn token_manager_remove_flow_limiter(
+        &self,
+        sc_address: ManagedAddress,
+        flow_limiter: ManagedAddress,
+    ) {
+        self.token_manager_proxy(sc_address)
+            .remove_flow_limiter(flow_limiter)
+            .execute_on_dest_context::<()>();
+    }
+
+    fn token_manager_add_flow_limiter(
+        &self,
+        sc_address: ManagedAddress,
+        flow_limiter: ManagedAddress,
+    ) {
+        self.token_manager_proxy(sc_address)
+            .add_flow_limiter(flow_limiter)
+            .execute_on_dest_context::<()>();
+    }
+
+    fn token_manager_transfer_operatorship(
+        &self,
+        sc_address: ManagedAddress,
+        operator: ManagedAddress,
+    ) {
+        self.token_manager_proxy(sc_address)
+            .transfer_operatorship(operator)
+            .execute_on_dest_context::<()>();
+    }
+
+    fn token_manager_is_minter(&self, sc_address: ManagedAddress, minter: &ManagedAddress) -> bool {
+        self.token_manager_proxy(sc_address)
+            .is_minter(minter)
+            .execute_on_dest_context()
+    }
+
+    fn esdt_get_token_properties(
+        &self,
+        token_identifier: EgldOrEsdtTokenIdentifier,
+        callback: CallbackClosure<Self::Api>,
+    ) {
+        let esdt_system_sc_address =
+            ESDTSystemSmartContractProxy::<Self::Api>::new_proxy_obj().esdt_system_sc_address();
+
+        let mut contract_call = self.send().contract_call::<()>(
+            esdt_system_sc_address,
+            ManagedBuffer::from("getTokenProperties"),
+        );
+        contract_call.push_raw_argument(token_identifier.into_name());
+
+        contract_call
+            .async_call()
+            .with_callback(callback)
+            .call_and_exit();
+    }
+
+    #[view]
+    #[storage_mapper("interchain_token_service")]
+    fn interchain_token_service(&self) -> SingleValueMapper<ManagedAddress>;
+
+    #[proxy]
+    fn interchain_token_service_proxy(
+        &self,
+        sc_address: ManagedAddress,
+    ) -> interchain_token_service::Proxy<Self::Api>;
+
+    #[proxy]
+    fn token_manager_proxy(
+        &self,
+        sc_address: ManagedAddress,
+    ) -> token_manager::Proxy<Self::Api>;
+
+    // TODO: Test that this callback works properly (probably only on devnet)
+    #[callback]
+    fn deploy_remote_token_callback(
+        &self,
+        salt: Hash<Self::Api>,
+        destination_chain: ManagedBuffer,
+        token_symbol: ManagedBuffer,
+        minter_raw: &ManagedBuffer,
+        gas_value: BigUint,
+        caller: ManagedAddress,
+        #[call_result] result: ManagedAsyncCallResult<MultiValueEncoded<ManagedBuffer>>,
+    ) {
+        match result {
+            ManagedAsyncCallResult::Ok(values) => {
+                let vec: ManagedVec<ManagedBuffer> = values.into_vec_of_buffers();
```

Comment:
It is on purpose since I only need to get specific values by index from the ManagedVec, which is contains the values returned by the `getTokenProperties` view of the ESDT System SC.
For reference where the indexes come from:
https://github.com/multiversx/mx-chain-go/blob/master/vm/systemSmartContracts/esdt.go#L1301